src/imperative/imperative_utils.h

```diff
@@ -575,12 +581,12 @@ inline void PushOperator(const OpStatePtr& state,
           [=](RunContext rctx) {
             run(rctx, engine::CallbackOnComplete());
           }, ctx, read_vars, write_vars, FnProperty::kNormal,
-          0, op->name.c_str());
+          0, *iter != "" ? iter->c_str() : op->name.c_str()); // huhanpeng: modified for profiling
     } else {
       CHECK(exec_type == ExecType::kAsync);
       Engine::Get()->PushAsync(
           run, ctx, read_vars, write_vars, FnProperty::kAsync,
-          0, op->name.c_str());
+          0, *iter != "" ? iter->c_str() : op->name.c_str()); // huhanpeng: modified for profiling
```

Comment:
You could replace the comment with explaining what it does, e.g. "append attribute name to make profiling unique" or whatever :)

Also generally, we try to avoid people's names within the code unless it's a bug/Todo. Feel free to add yourself to the contributors.md!

Reply:
Yes definitely add your name here - https://github.com/apache/incubator-mxnet/blob/master/CONTRIBUTORS.md